File: contracts/csharp/TerminalCommand.cs
namespace HackerDashboard.Contracts;

/// <summary>A terminal line parsed into a verb and its arguments, sent to the backend over HTTP.</summary>
/// <param name="Raw">Exactly what the user typed — kept for terminal history.</param>
/// <param name="Verb">The action, e.g. "theme", "toggle", "prompt", "fetch".</param>
public sealed record TerminalCommand(
    string Raw,
    string Verb,
    IReadOnlyDictionary<string, string> Args);

File: contracts/typescript/TerminalCommand.ts
/** A terminal line parsed into a verb and its arguments, sent to the backend over HTTP. */
export interface TerminalCommand {
  /** Exactly what the user typed — kept for terminal history. */
  raw: string;
  /** The action, e.g. "theme", "toggle", "prompt", "fetch". */
  verb: string;
  args: Record<string, string>;
}
